internal/ns/resource_provision_vpx.go

```diff
@@ -1388,104 +1287,67 @@ func networkinterfaceTonetworkinterfaceTF(nif []interface{}, stateNetworkInterfa
-	var nifs []map[string]interface{}
-
-	// if network_interfaces is not set in the state file, then return empty list
-	if len(stateNetworkInterface.Elements()) == 0 {
-		return basetypes.ListValue{}
-	}
+	var nifListFilterFromGetResponse []map[string]interface{}
 
-	for _, v := range stateNetworkInterface.Elements() {
+	for _, v := range nifGetResponse {
 
-		inputNifs := v.(basetypes.ObjectValue).Attributes()
-
-		// get the portnames of all the inputNifs
-		var inputNifPortNames []string
-		for key, inputNif := range inputNifs {
-			if key == "port_name" {
-				inputNif := inputNif.(basetypes.StringValue).ValueString()
-				inputNifPortNames = append(inputNifPortNames, inputNif)
-			}
-		}
-
-		for _, nif := range nif {
-			nifMap := nif.(map[string]interface{})
-			if len(nifMap) == 0 {
-				continue
-			}
-			if !service.Contains(inputNifPortNames, nifMap["port_name"].(string)) {
-				continue
-			}
-			// iterate through nifMap and only keep the keys that are in the nifSchemaAttributes
-			// var nifMap2 map[string]interface{}
-			nifMap2 := make(map[string]interface{})
-			for k, v := range nifMap {
-				if service.Contains(nifSchemaAttributes, k) {
-					nifMap2[k] = v
+		nifMapGetResponse := v.(map[string]interface{})
+		nifMapFilterFromGetResponse := make(map[string]interface{})
+		if nifMapGetResponse["port_name"].(string) != "" && nifMapGetResponse["parent_channel_id"].(string) == "" {
+			for key, inputNif := range nifMapGetResponse {
+				if service.Contains(nifSchemaAttributes, key) {
+					nifMapFilterFromGetResponse[key] = inputNif
 				}
 			}
-			nifs = append(nifs, nifMap2)
+
+			nifListFilterFromGetResponse = append(nifListFilterFromGetResponse, nifMapFilterFromGetResponse)
 		}
 	}
+
 	var nifList []attr.Value
-	for listindex, nif := range nifs {
-		var networkInterfaceElementsInState []string
-		stateNetworkInterfaceElements := stateNetworkInterface.Elements()[listindex].(basetypes.ObjectValue)
-		for key, val := range stateNetworkInterfaceElements.Attributes() {
-			if !val.IsNull() {
-				networkInterfaceElementsInState = append(networkInterfaceElementsInState, key)
-			}
-		}
-		var newMap basetypes.ObjectValue
+	for _, nif := range nifListFilterFromGetResponse {
+
+		var nifObjectValue basetypes.ObjectValue
 		nifmap := make(map[string]attr.Value)
 		nifMapTypes := make(map[string]attr.Type)
 		for k, v := range nif {
-			if !service.Contains(networkInterfaceElementsInState, k) {
-				if k == "is_vlan_applied" || k == "is_mgmt_ifc" || k == "is_member_ifc" || k == "l2_enabled" || k == "receiveuntagged" {
-					nifmap[k] = basetypes.NewBoolNull()
-					nifMapTypes[k] = types.BoolType
-				} else if k == "vlan" {
-					nifmap[k] = basetypes.NewInt64Null()
-					nifMapTypes[k] = types.Int64Type
-				} else if k == "vrid_list_ipv4_array" || k == "vrid_list_ipv6_array" || k == "vlan_whitelist_array" {
+
+			if k == "is_vlan_applied" || k == "is_mgmt_ifc" || k == "is_member_ifc" || k == "l2_enabled" || k == "receiveuntagged" {
```

Comment:
Try to make this logic generic. Check if we can pass `k` to a function and get it's type from the Schema.